hedera-node/hedera-app/src/main/java/com/hedera/node/app/metrics/NodeMetrics.java

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.node.app.metrics;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.state.roster.RosterEntry;
+import com.swirlds.common.metrics.RunningAverageMetric;
+import com.swirlds.metrics.api.DoubleGauge;
+import com.swirlds.metrics.api.Metrics;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@Singleton
+public class NodeMetrics {
+    private static final String APP_CATEGORY = "app_";
+    private static final Logger log = LogManager.getLogger(NodeMetrics.class);
+    private final Map<Long, RunningAverageMetric> activeRoundsAverages = new HashMap<>();
+    private final Map<Long, DoubleGauge> activeRoundsSnapshots = new HashMap<>();
+    private final Metrics metrics;
+
+    @Inject
+    public NodeMetrics(@NonNull final Metrics metrics) {
+        this.metrics = requireNonNull(metrics);
+    }
+
+    /**
+     * Registers the metrics for the active round % for each node in the given roster.
+     *
+     * @param rosterEntries the list of roster entries
+     */
+    public void registerMissingNodeMetrics(@NonNull List<RosterEntry> rosterEntries) {
```

Comment:
`Missing`–are these metrics we somehow removed, or are they new metrics? Just curious what `missing` means in this case

Ah, does this mean the nodes that did _not_ participate (i.e. were missing) from consensus for a given round?

Reply:
I just thought when nodes are added or removed. Will change name to `registerNodeMetrics`

Reply:
Changed. Thanks @mhess-swl